crates/factors/src/factor.rs

```diff
@@ -7,15 +7,20 @@ use crate::{
     InstanceBuilders, Linker, PrepareContext, RuntimeConfigSource, RuntimeFactors,
 };
 
+/// A contained (i.e., "factored") piece of runtime functionality.
 pub trait Factor: Any + Sized {
```

Comment:
I wrote better docs for this trait [in the SIP](https://github.com/fermyon/spin/pull/2518/files#diff-3ea0e4a724fed06ab763f915e6cdcbbabf8c84bc63e90882cd6da1515b55a768R74) which could be adapted.